### PR TITLE
Generate base project and UI

### LINKED ISSUE #1: Generate basic framework
Create a new React application and Node server.

### PR DESCRIPTION
**Pull request by nvanselow**
Creates a mono-repo for the project and generates a React UI with Create React App.

Closes #1 

- Yarn workspaces for monorepo (Lerna is probably overkill for this). Use yarn 1.13.0 for best results.
- Prettier and Eslint (with precommit hooks)
- React + Typescript with Create React App
- React Emotion
- [React Testing Library](https://github.com/kentcdodds/react-testing-library) instead of Enzyme

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
```
Comment (MrMonsk) [inline .eslintrc.json]: I've got a six-month old eslint config that I've liked, though will need to edit to do ts parsing. IIRC we will still need `typescript-eslint-parser` and `eslint-plugin-typescript`

```javascript
module.exports = {
  extends: ['airbnb', 'prettier', 'prettier/react'],
  plugins: ['prettier'],
  parser: 'babel-eslint',
  env: {
    browser: true,
  },
  rules: {
    // general js
    'import/default': 1,
    'import/named': 1,
    'import/namespace': [1, {allowComputed: true}],
    'import/prefer-default-export': 0,
    'import/no-extraneous-dependencies': [1, {devDependencies: true}],
    'import/order': [1, {'newlines-between': 'always'}],
    'no-confusing-arrow': 0,
    'no-underscore-dangle': 0,

    // prettier
    'prettier/prettier': [
      2,
      {
        printWidth: 100,
        tabWidth: 4,
        semi: false,
        singleQuote: true,
        trailingComma: 'es5',
        bracketSpacing: false,
      },
    ],
    
    // react-specific
    'react/require-default-props': 0,
    'react/no-unused-prop-types': 0,
    'react/jsx-filename-extension': 0,

    // custom
    'jsx-a11y/anchor-is-valid': [ 'error', {
      'components': ['Link'],
      'specialLink': ['to'],
      'aspects': ['noHref', 'invalidHref', 'preferButton']
    }],
  },
  settings: {
    'import/ignore': ['node_modules', '.(png|svg|jpg|css|pdf)$'],
    'import/extensions': ['.js'],
  },
}```

**Review by MrMonsk — COMMENTED** [review]
Looks good, I am pretty opinionated on my eslint/prettier settings though, so if you want to take a look at the config I've pasted/comment on ts.

File: .eslintrc.json (hunk 1 above)
Comment (nvanselow) [reply .eslintrc.json]: Sure. I don't have strong preferences as long as it is consistent. No semi colons is going to take a bit of getting used to, but I'm up for trying it.

Comment (nvanselow) [reply .eslintrc.json]: Lint settings updated!

File: .prettierrc.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+module.exports = {
+    printWidth: 100,
+    tabWidth: 4,
```
Comment (MrMonsk) [inline .prettierrc.js]: O shoot, we should make this 2, I forgot the project that I pasted this from was 4 for some reason.

**Review by MrMonsk — APPROVED** [review]
Looks good, just change tabWidth back to 2, that was my bad copy and paste from prev project :)

File: .prettierrc.js (hunk 2 above)
Comment (nvanselow) [reply .prettierrc.js]: Done

**Pull request merged**